### PR TITLE
Update to Jetpack 5.1.2 / L4T 35.4.1

### PR DESCRIPTION
**Pull request by danielfullmer**
###### Description of changes

Jetpack release notes: https://docs.nvidia.com/jetson/archives/jetpack-archived/jetpack-512/release-notes/index.html
This release mainly brings support for the Orin AGX Industrial variant, as well as multi-gigabit support in UEFI on the Orin AGX devkits.

---

Following the checklist from `UPGRADE_CHECKLIST.md`:
- [x] Update `l4tVersion`, `jetpackVersion`, and `cudaVersion` in default.nix
- [x] Update branch/revision/sha256s in:
    - [x] `default.nix`
    - [x] `kernel/default.nix`
    - [x] `kernel/display-driver.nix`
    - [x] `uefi-firmware.nix`
    - [x] `optee.nix`
    - [x] Grep for "sha256 = ", see if there is anything else not covered
- [x] Grep for the previous version strings e.g. "35.3.1"
- [x] Compare files from `unpackedDebs` before and after
- [x] Ensure the soc variants in device-pkgs.nix match those in jetson_board_spec.cfg from BSP
- [ ] Ensure logic in ota-utils/ota_helpers.func matches nvidia-l4t-init/opt/nvidia/nv-l4t-bootloader-config.sh

###### Testing

Tested that:
- [x] Run `nix flake check`
- [ ] Build installer ISO
- [x] Flash all variants
- [x] Boot all variants
- [x] Run our (Anduril's) internal automated device tests

### CODE REVIEW

**Comment by danielfullmer** [comment]
Added support for the "AGX Orin Industrial" module.  This can be enabled by setting `som = "orin-agx-industrial";`

Tested by flashing and booting NixOS on a AGX Orin Industrial (p3701-0008) + Orin Devkit carrier board (p3767-0000).
```
[root@orin-agx-industrial-devkit]# hostnamectl
... [ trimmed ] ...

  Hardware Model: Jetson AGX Orin Industrial
Firmware Version: 35.4.1
```

File: flash-tools.nix [hunk 1]
```diff
@@ -40,14 +33,12 @@ let
       # We should never be flashing upstream's kernel, so just remove it so we get errors if it is used
       #rm -f kernel/Image*
 
-      # Flash script looks for this file
+      # Remove the big nv_tegra dir, since its not neede by flash scripts.
```
Comment (mikatammi) [inline flash-tools.nix:36]: Spelling

**Comment by mikatammi** [comment]
Should this PR also include the https://developer.nvidia.com/downloads/embedded/l4t/r35_release_v4.1/overlay_jaoi_low_temp_35.4.1.tar.gz ? 

"This overlay resolves a low temp boot issue with the Jetson AGX Orin Industrial Module. "

File: flash-tools.nix (hunk 1 above)
Comment (mikatammi) [reply flash-tools.nix:36]: "neede"

**Comment by mikatammi** [comment]
Also there are some Vulkan samples, maybe they could come in separate commit: https://developer.nvidia.com/downloads/embedded/l4t/r35_release_v4.1/vulkan-sc-samples.zip "This file contains the top level cmake files used to build the Vulkan SC Samples. "

**Comment by danielfullmer** [comment]
> Should this PR also include the https://developer.nvidia.com/downloads/embedded/l4t/r35_release_v4.1/overlay_jaoi_low_temp_35.4.1.tar.gz ?
> 
> "This overlay resolves a low temp boot issue with the Jetson AGX Orin Industrial Module. "

Added this overlay in a recent push.

File: flash-tools.nix (hunk 1 above)
Comment (danielfullmer) [reply flash-tools.nix]: Thanks! Fixed in latest push

**Comment by gerasiov** [comment]
@danielfullmer @mikatammi 
Hi, folks. Is there any news on this PR?

**Comment by danielfullmer** [comment]
@gerasiov I intend to merge this shortly after some final testing. This PR has been open far too long...

**Comment by danielfullmer** [comment]
Just a status update: This is currently blocked on the multimedia-samples test not passing. It segfaults when running: https://github.com/anduril/jetpack-nixos/blob/master/pkgs/samples/default.nix#L274-L298

**Comment by mikatammi** [comment]
There's already version r35.5.0 GA available: https://docs.nvidia.com/jetson/archives/r35.5.0/DeveloperGuide/index.html

I wonder if it fixes some of the issues like the multimedia-samples test. This GA (General Availability) Release should be production-ready by NVIDIA's standards.

(Also there has been some developer preview for r36.2 https://docs.nvidia.com/jetson/archives/r36.2/spe/index.html

But it could be nice if the stable release r35.5.0 GA could be also supported even if this DP version is added.)

**Comment by danielfullmer** [comment]
@mikatammi Finally fixed the multimedia-samples test now working with the following fix: https://github.com/anduril/jetpack-nixos/pull/119/commits/205ae5c8bef666043d813e2dc326f8ca07084e57

Thanks for the reminder that 35.5.0 was released. We're looking into that now.  I did see the Jetpack 6 DP release, and it looks interesting given the "bring our own kernel" thing they now allow.  It's not clear when I'll personally have time to look into it in detail, but it's on my radar.  Since I'll need to support Xavier devices for the forseeable future, we'll have to keep support for Jetpack 5, but hopefully we can find a way to have them both work well without forking.

**Pull request merged**